src/XrdHttp/XrdHttpProtocol.cc

```diff
@@ -685,6 +697,47 @@ int XrdHttpProtocol::Process(XrdLink *lp) // We ignore the argument here
 
   }
 
+
+  if (postheaderwait) {
+    postheaderwait = false;
+    if (SSL_verify_client_post_handshake(ssl) != 1) {
+      // This is hit if the remote client doesn't support the post-handshake authentication
+      // (curl, Mac OSX) or TLS v1.3 (RHEL7).
+      TRACEI(ALL, "Unable to request client X.509 authentication");
+      ERR_print_errors(sslbio_err);
```

Comment:
Should this have a `return -1;`? Without it, the code will treat it as authentication succeeded for the requested prefixes even when the remote client doesn't support the post-handshake authentication.